### PR TITLE
Add Qwen register attention implemention

### PR DESCRIPTION
**Pull request by leonardozcm**
## Description

Customized Qwen in https://github.com/intel-analytics/ipex-llm/issues/11103 is passing `registered_causal_mask` cross attention forward.

```
-------------------- Prompt --------------------
AI是什么？
-------------------- Output --------------------
AI是什么？"
AI是计算机科学的一个分支，是一门模拟、延伸和扩展人的智能的科学。它涵盖了机器学习、自然语言处理、计算机视觉、
```

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/convert.py [hunk 1]
```diff
@@ -1510,7 +1517,7 @@ def safe_bmm_fwd(*args, **kwargs):
         from ipex_llm.transformers.models.phi import model_forward
         convert_forward(model, module.PhiAttention, attention_forward)
         convert_forward(model, module.PhiModel, model_forward)
-    elif model.config.model_type in ["phi3", "phi3_v"]:
+    elif model.config.model_type == "phi3":
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/convert.py:1520]: why remove "phi3-v" here ?

Comment (leonardozcm) [reply python/llm/src/ipex_llm/transformers/convert.py:1520]: oh it is a typo

**Review by rnwang04 — APPROVED** [review]
others LGTM.
Just a question here，the demo output seems a little unreasonable, maybe have a double check.

File: python/llm/src/ipex_llm/transformers/models/qwen.py [hunk 2]
```diff
@@ -180,6 +180,132 @@ def qwen_attention_forward(
...
+    else:
+        if q_len > 1:
+            causal_mask = registered_causal_mask[
+                :, :, kv_seq_len - q_len:kv_seq_len, :kv_seq_len
+            ]
+            attention_mask = torch.zeros(causal_mask.shape, dtype=query_states.dtype,
+                                         device=query_states.device)
+            attention_mask.masked_fill_(causal_mask.logical_not(),
+                                        torch.finfo(attention_mask.dtype).min)
+            attention_mask = attention_mask.expand([bsz, -1, -1, -1])
```
[the remark this answers is not in the document]
Comment (leonardozcm) [reply python/llm/src/ipex_llm/transformers/models/qwen.py:270]: Yes it is for both.

**Pull request merged**